[Feature] 플로팅 버튼 구현

sihyun92 opened the pull request:
### 📌 개요

<!-- PR과 관련있는 Issue를 closed 해주세요 -->

closed #87 

작업한 내용에 대해 간단히 작성해주세요.

### ✅ 작업내용

- 플로팅 버튼 구현
- 팀 별 플로팅 버튼 구현
- 플로팅 버튼 클릭 시 말풍선 구현
- 경기 종료 시 채팅 플로팅 비활성화 구현

### 📝 공유사항

팀원들에게 공유할 사항이나 변경 사항들을 작성해주세요.

- 현재 채팅방 플로팅 버튼을 클릭하고, 다른 채팅방을 클릭했을 때 처음에 클릭한 채팅방의 말풍선이 없어지지 않는 버그가 있습니다. 저랑 같이 해결해봤으면 좋겠습니다.
- 애니메이션은 일단 추후에 추가할 예정입니다.

sihyun92 committed:
Chore: 채팅 로고 추가

sihyun92 committed:
Feature: 채팅방 플로팅 버튼 구현
- 경기별 채팅방 플로팅 버튼 구현
- 클릭 시 말풍선 메시지 구현

현재 채팅방을 클릭 한 상태에서 다른 채팅방을 눌렀을 시 첫번째 채팅방의 말풍선이 없어지지 않는 버그 발생! 다른 버그는 없음

sihyun92 committed:
Merge branch 'dev' of github.com:baseball-project/client into feature/today

sihyun92 committed:
Design: 플로팅버튼 UI 변경

doitidey reviewed (commented):
작업 수고하셨습니다!
코드를 살펴봤는데, `chat.svg`를 사용하는 기본 플로팅 버튼이 떠 있어야하는데 그 부분이 누락 되어있는 것 같아요! 플로팅 버튼을 호출하는 코드도 추가해주셔야 할 것 같습니다~@!

doitidey commented on src/components/float/Float.tsx:
이 컴포넌트는 비어있고 호출되지도 않는 것 같은데, 어떤 목적으로 사용되는 건지 궁금합니다! 🤔

sihyun92 replied:
죄송합니다 ㅜㅜ Today 컴포넌트에 Float 컴포넌트 적용했는데 깜빡하고 pr을 안올려드렸습니다 ㅜㅜ 참고로 이 컴포넌트는 플로팅 버튼을 클릭했을 시 투명처리된 검은색 백그라운드로 사용되는 컴포넌트입니다

doitidey commented on src/components/float/ChatRooms.tsx:
```diff
@@ -0,0 +1,28 @@
+function ChatRooms({ game, visibleFloat }: ChatRoomsProps) {
+  return (
+    <ul className="chat-rooms-block">
+      {game.map((item) => (
```
게임이 없는 날도 있어서, 예외 처리가 필요해보입니다! 메인 컴포넌트가 있다면 그 곳에서 진행되면 좋을 것 같습니다 :)

sihyun92 replied:
이 부분 예외처리 해놓도록 하겠습니다~!! 감사합니다!!

doitidey commented on src/components/float/ChatRoomItem.tsx:
```diff
@@ -0,0 +1,48 @@
+
+  return (
+    <li className="item-block">
+      {entrance && (
+        <div className="entrance">
+          <Text small>채팅방에 입장할래요?</Text>
+          <div className="entrance__button">
+            <span>네</span>
+            <span onClick={onCancel}>취소</span>
+          </div>
+        </div>
+      )}
```
입장 팝업이 `ChatRoomItem`마다 작성되어 있고, `entrance`가 버튼마다 참조를 하기 때문에 다른 버튼에서 클릭이 일어나도 해당 팝업이 사라지지 않는 버그가 생긴 것 같습니다. 이를 해결하기 위해서는 전역 상태 관리를 통해 on/off 기능을 구현하는 것이 필요할 것 같습니다. `useModal` hook을 참고하셔도 좋을 것 같습니다!😃

sihyun92 committed:
Feat: 플로팅 버튼 UI 적용

sihyun92 committed:
Fix: Today 컴포넌트 적용

sihyun92 committed:
Feat: favicon 적용

sihyun92 committed:
Fix: 경기 종료시 플로팅 버튼 비활성화 예외처리 완료

sihyun92 committed:
Fix: disabled 수정

doitidey committed:
Refactor: float 컴포넌트들 로직 분리 및 전체적인 리팩토링

- today에서 진행하는 대신 chatFloatSection을 만들어 관리하고 의존성을 분리
- FloatButton에서 button 스타일에 따라 다른 스타일의 버튼을 출력하도록 함
- 채팅룸 출력 버튼과 팝업 컴포넌트 분리 및 클릭된 채팅룸의 id 번호에 따라 팝업 출력

doitidey committed:
Refactor: visiblefloat 옵셔널 제거

doitidey committed:
Refactor: Button boolean 속성 사용 string으로 변경

doitidey committed:
Feature: 팝업 오픈 전역함수 연동
